### PR TITLE
[DEV-1867] Auto create buyCrypto checkout

### PR DESCRIPTION
**Pull request by Yannick1712**
#### Pre-Release

- [ ] Check migrations
  - No database related infos (sqldb-xxx)
  - Impact on GS (new/removed columns)
- [ ] Check for linter errors (in PR)
- [ ] Test basic user operations
  - Login/logout
  - Registration
  - Update user data
  - Create/delete payment routes

#### Post-Release

- Test basic user operations
- Monitor application insights log

### CODE REVIEW

**Comment by Yannick1712** [comment]
Add Migration

File: src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts [hunk 1]
```diff
@@ -69,15 +71,29 @@ export class BuyCryptoService {
...
+  async createFromBankTx(bankTx: BankTx, buyId: number): Promise<BuyCrypto> {
+    let entity = await this.buyCryptoRepo.findOneBy({ bankTx: { id: bankTx.id } });
     if (entity) throw new ConflictException('There is already a buy-crypto for the specified bank TX');
 
     entity = this.buyCryptoRepo.create();
 
     // bank tx
-    entity.bankTx = await this.bankTxService.getBankTxRepo().findOneBy({ id: bankTxId });
-    if (!entity.bankTx) throw new BadRequestException('Bank TX not found');
+    entity.bankTx = bankTx;
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts:81]:
```ts
entity = this.buyCryptoRepo.create({ bankTx });
```

File: src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts [hunk 2]
```diff
@@ -69,15 +71,29 @@ export class BuyCryptoService {
...
+  }
+
+  async createFromCheckoutTx(checkoutTx: CheckoutTx, buyId: number): Promise<BuyCrypto> {
+    let entity = await this.buyCryptoRepo.findOneBy({ checkoutTx: { id: checkoutTx.id } });
+    if (entity) throw new ConflictException('There is already a buy-crypto for the specified checkout TX');
+
+    entity = this.buyCryptoRepo.create();
+
+    // checkout tx
+    entity.checkoutTx = checkoutTx;
```
Comment (davidleomay) [inline src/subdomains/core/buy-crypto/process/services/buy-crypto.service.ts:96]:
```ts
entity = this.buyCryptoRepo.create({ checkoutTx });
```

File: src/subdomains/supporting/fiat-payin/services/checkout-tx.service.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    // send internal Mail
+    await this.notificationService.sendMail({
+      type: MailType.ERROR_MONITORING,
+      input: {
+        subject: 'Checkout Tx Error - BankUsage not matching format',
+        errors: [`CheckoutTx Id: ${tx.id}; Buy Route for bankUsage ${tx.description} not matching`],
+      },
+    });
+  }
```
Comment (davidleomay) [inline src/subdomains/supporting/fiat-payin/services/checkout-tx.service.ts:49]: I think a single mail for both cases should be enough.
```ts
      // send error mail
      await this.notificationService.sendMail({
        type: MailType.ERROR_MONITORING,
        input: {
          subject: 'Invalid Checkout Tx',
          errors: [`No buy route found for CheckoutTx with ID ${tx.id} and bankUsage ${tx.description}`],
        },
      });
```

Comment (davidleomay) [reply src/subdomains/supporting/fiat-payin/services/checkout-tx.service.ts:49]: Add an error log

**Review by davidleomay — APPROVED** [review]

**Review by davidleomay — APPROVED** [review]

**Pull request merged**